Update limits in death_match_manager

Specifer opened the pull request:
Does this solve issue #771?

Specifer committed:
Update death_match_manager.js

Does this solve the issue #771 ?

OttoRocket-LVP commented:
Hello Specifer,

Thank you for your contribution!

This might indeed be what will solve the linked issue. Can you please test it on your local server?

I believe you'll also want to add a wait(0).then(() => player.updateStreamer...) to avoid the reentrancy warnings.

OttoRocket-LVP commented on javascript/features/death_match/death_match_manager.js:
```diff
@@ -214,6 +214,10 @@ export class DeathMatchManger {
 
+        // Force-update the streamer for the player, based on where they will be spawning.
+        player.updateStreamer(spawnPosition.position, location.world,
+        location.interiorId, /* STREAMER_TYPE_OBJECT= */ 0);
```
NIT: You'll want to add an extra tab (indenting) here

Specifer replied:
it's not working :(

Specifer replied:
Callback (98) ignored because a ScopedIgnore is in place.
[23:10:38] [WARNING][callback_hook.cc:80] Callback (OnPlayerActivityChange) ignored << because a ScopedIgnore is in place.
[23:10:38] [WARNING][callback_hook.cc:80] Callback (OnPlayerActivityChange) ignored << because a ScopedIgnore is in place.

OttoRocket-LVP replied:
How about you restrict a player from using the command when they're in a car? :)

Specifer replied:
yeah. I think that's the only way now.

Specifer committed:
Update death_match_manager.js

Holsje, I couldn't check this if it is working though. There is a problem with my local server :(. Could you say if your's work and this works on your local one?

Specifer commented:
Holsje?

OttoRocket-LVP commented:
Heya, thank you for the modifications! :)
Did you test it on your local server?

Would also be nice if you'd add a tab at line 219

Specifer committed:
Update death_match_manager.js

Specifer commented:
It's not working :(

 

Callback (98) ignored because a ScopedIgnore is in place.
[23:10:38] [WARNING][callback_hook.cc:80] Callback (OnPlayerActivityChange) ignored << because a ScopedIgnore is in place.
[23:10:38] [WARNING][callback_hook.cc:80] Callback (OnPlayerActivityChange) ignored << because a ScopedIgnore is in place.

Specifer committed:
Update death_match_manager.js

Specifer committed:
Update death_match_commands.js

Specifer committed:
Merge branch 'deathmatch-update-streamer' of https://github.com/Specifer/playground into deathmatch-update-streamer

Specifer commented:
@OttoRocket-LVP. Done! and tested.

OttoRocket-LVP commented:
Thanks for the update!

There's a `limits` class you should use for this instead.

Also there's limit checks at `enterDeathMatchZone` I believe it would fit there better.

Specifer committed:
Update death_match_commands.js

Specifer committed:
Revert "Update death_match_commands.js"

This reverts commit 9a9224ecd00b61bb2b9304032782a1b6acf560c3.

Specifer committed:
Corrected

Is this what you were asking?

Specifer committed:
Corrected Fully

Specifer committed:
Left out unsaved file

Specifer committed:
Update messages.json

Specifer commented:
There are some test errors since I didn't modify the test, which I will apart from that any probs?

OttoRocket-LVP commented:
Try running the tests again. I believe some gang related are somewhat involved with a random and thus not consistent. Something we should look into.

OttoRocket-LVP commented on javascript/features/limits/limits_decider.js:
```diff
@@ -150,6 +154,15 @@ export class LimitsDecider extends PlayerEventObserver {
+   // Processes if the player is present in vehicle and makes sure if the player isn't in one.
+   // This proces is required to avoid uncertain bugs .
+    processVehicleRequirement(player) {
+        if(player.state == Player.kStateVehicleDriver && Player.kStateVehiclePassenger)
```
I don't think this if statement is doing what you want.

Currently it states: If the player's state equals to `Player.kStateVehicledriver` AND the value of `Player.kStateVehiclePassenger` is truthy.

The latter is _always_ true because it's a static value. I assume you want a || `player.state === Player.kStateVehiclePassenger`

Also please test this as passenger. 
Also please try to use `===` instead of `==`.

OttoRocket-LVP commented on javascript/features/limits/limits_decider.js:
```diff
@@ -150,6 +154,15 @@ export class LimitsDecider extends PlayerEventObserver {
+   // This proces is required to avoid uncertain bugs .
+    processVehicleRequirement(player) {
+        if(player.state == Player.kStateVehicleDriver && Player.kStateVehiclePassenger)
+        return LimitsDecision.createRejection(Message.LIMITS_DEATH_MATCH_ENTRY_VEHICLE);
```
Nit: Please add a tab

OttoRocket-LVP replied:
This message is not DEATH_MATCH related anymore but a generic limits_decider message.

Please follow message naming convention according to the other deciders.

OttoRocket-LVP commented on javascript/features/death_match/death_match_commands.js:
```diff
@@ -47,8 +48,15 @@ export class DeathMatchCommands {
+        if (canEnterDeathmatchZone.isApproved()) {
+            player.sendMessage(Message.LIMITS_DEATH_MATCH_ENTRY_VEHICLE, canEnterDeathmatchZone);
+            return;
+            }
```
Remove tab

OttoRocket-LVP commented on javascript/features/death_match/death_match_commands.js:
```diff
@@ -47,8 +48,15 @@ export class DeathMatchCommands {
         // If the |player| is switching to a new deathmatch zone, we allow them to switch when they
-        // haven't been shot recently. Otherwise we need to meet the full minigame criteria.
+        // haven't been shot recently or if they aren't in vehicle. Otherwise we need to meet the full minigame criteria.
+        if (canEnterDeathmatchZone.isApproved()) {
```
If it's approved I believe you _don't_ want to be able to enter the DM zone.

Are you sure you tested it?
